x/merkledb/db.go

```diff
@@ -1242,12 +1269,43 @@ func (db *merkleDB) getNode(key Key, hasValue bool) (*node, error) {
 	switch {
 	case db.closed:
 		return nil, database.ErrClosed
-	case key == Key{}:
-		return db.root, nil
 	case hasValue:
 		return db.valueNodeDB.Get(key)
+	default:
+		return db.intermediateNodeDB.Get(key)
+	}
+}
+
+func (db *merkleDB) getRoot() maybe.Maybe[*node] {
+	return db.root
+}
+
+func (db *merkleDB) Clear() error {
+	db.commitLock.Lock()
+	defer db.commitLock.Unlock()
+
+	// TODO is it faster to clear another way?
+	keysToDelete, err := db.getKeysNotInSet(maybe.Nothing[[]byte](), maybe.Nothing[[]byte](), set.Set[string]{})
+	if err != nil {
+		return err
+	}
+
+	ops := make([]database.BatchOp, len(keysToDelete))
+	for i, keyToDelete := range keysToDelete {
+		keyToDelete := keyToDelete
+		ops[i] = database.BatchOp{
+			Key:    keyToDelete,
+			Delete: true,
+		}
 	}
-	return db.intermediateNodeDB.Get(key)
+
+	// Don't need to lock [view] because nobody else has a reference to it.
+	view, err := newTrieView(db, db, ViewChanges{BatchOps: ops})
+	if err != nil {
+		return err
+	}
+
+	return view.commitToDB(context.Background())
 }
```

Comment:
Shouldn't we just be wiping the intermediate nodes and then wiping the value nodes? Or are we trying to allow someone to call this and maintain the change in the history?